src/plugins/Polls/UI/Polls.tsx

```diff
@@ -0,0 +1,115 @@
+            margin: theme.spacing(1, 0),
+            padding: theme.spacing(0, 1),
+            height: '28px',
+        },
+        bar: {
+            position: 'absolute',
+            top: '0',
+            left: '0',
+            zIndex: 100,
+            backgroundColor: theme.palette.primary.main,
+            opacity: 0.6,
+            minWidth: theme.spacing(1),
+            height: '28px',
+            borderRadius: theme.spacing(0.8),
+        },
+        text: {
+            zIndex: 101,
+            lineHeight: '28px',
+            margin: '0 4px',
+        },
+        deadline: {
+            color: '#657786',
+        },
+    }),
+)
+
+interface PollCardProps {
+    poll: PollGunDB
+    onClick?(): void
+    vote?(poll: PollGunDB, index: number): void
+    status?: PollStatus
+}
+
+export type PollStatus = 'Voted' | 'Voting' | 'Error' | 'Closed' | 'Inactive'
+
+export function PollCardUI(props: PollCardProps) {
+    const { poll, onClick, vote, status } = props
+    const classes = useStyles()
+    const isClosed = new Date().getTime() > poll.end_time ? true : false
+
+    const totalVotes = poll.results.reduce(
+        (accumulator: number, currentValue: number): number => accumulator + currentValue,
+    )
+
+    return (
+        <Card className={classes.card} onClick={() => onClick?.()}>
+            <div className={classes.line}>
+                <div style={{ fontSize: '16px' }}>{poll.question}</div>
+                {!status || status === 'Inactive' ? null : (
+                    <div className={classes.status}>
+                        {status === 'Voting' ? <CircularProgress size={18} /> : null}
+                        <span className={classes.statusText}>{status}</span>
+                    </div>
+                )}
+            </div>
+            <div>
+                {poll.options.map((option, index) => (
+                    <div
+                        className={classes.option}
+                        key={index}
+                        onClick={() => {
+                            vote?.(poll, index)
+                        }}>
+                        <div
+                            style={{
+                                display: 'flex',
+                            }}>
+                            <div
+                                className={classes.bar}
+                                style={{
+                                    width: `${(poll.results[index] / totalVotes) * 100}%`,
+                                }}></div>
+                            <div className={classes.text}>{option}</div>
+                        </div>
+                        <div className={classes.text}>{poll.results[index]}</div>
+                    </div>
+                ))}
+            </div>
+            <Typography variant="body2" classes={{ root: classes.deadline }}>
+                {isClosed ? 'Closed' : `Deadline: ${dateFormat('YYYY-mm-dd HH:MM', new Date(poll.end_time))}`}
```

Comment:
You may want to find if the built-in `Date` object can do this kind of format for you instead of re-inventing the wheel